### PR TITLE
Disable service worker on remaining hardcoded redirects

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

The service worker doesn't play well with redirects - see #7156 - this disables it in the `/shop` redirect as well.

**update**: I also added all the remaining redirects in `config/routes.rb`

## Related Tickets & Documents

Closes https://github.com/thepracticaldev/dev.to/issues/7339

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

### CODE REVIEW

File: app/views/service_worker/index.js.erb [hunk 1]
```diff
@@ -126,10 +126,11 @@
         !url.href.includes('/search/users') &&
 
         // Don't run on redirects
-        !url.href.includes('/workshops') &&
         !url.href.includes('/%F0%9F%92%B8') && // 💸 (hiring)
-        !url.href.includes('/podcasts') &&
         !url.href.includes('/p/rlyweb') &&
+        !url.href.includes('/podcasts') &&
+        !url.href.includes('/shop') &&
+        !url.href.includes('/workshops') &&
```
Comment (Ridhwana) [inline app/views/service_worker/index.js.erb:133]: 💥+ 1 on the alphabetizing :D

Comment (rhymes) [reply app/views/service_worker/index.js.erb:133]: Sublime Text 3 -> Sort lines :D

File: app/views/service_worker/index.js.erb [hunk 2]
```diff
@@ -126,10 +126,11 @@
         !url.href.includes('/search/users') &&
 
         // Don't run on redirects
-        !url.href.includes('/workshops') &&
         !url.href.includes('/%F0%9F%92%B8') && // 💸 (hiring)
-        !url.href.includes('/podcasts') &&
         !url.href.includes('/p/rlyweb') &&
+        !url.href.includes('/podcasts') &&
+        !url.href.includes('/shop') &&
```
Comment (jitendra1998) [inline app/views/service_worker/index.js.erb:132]: can you fix it for dev.to/api as well

Comment (rhymes) [reply app/views/service_worker/index.js.erb:132]: great call! going to add it

**Comment by rhymes** [comment]
@Ridhwana sorry I re-requested your review as it morphed in a more generic PR

**Pull request merged**